db/migrate/20171231134017_change_vlan_to_int.rb

```diff
@@ -0,0 +1,11 @@
+class ChangeVlanToInt < ActiveRecord::Migration[5.1]
+  def up
+    Subnet.unscoped.where(vlanid: "").update_all(vlanid: nil)
+    opts = connection.adapter_name.downcase == 'postgresql' ? {using: 'vlanid::integer'} : {}
```

Comment:
why is this required? shouldnt the schema define this?

Reply:
Otherwise postgres throws an error, you need to tell it explicitly how you want to convert the string.

Reply:
thanks, any idea how the empty string got there to begin with? any place we should make sure we dont add new records as strings?

Reply:
When submitting a form it was saved as blank string instead of nil because that is the parameter that was passed to the server. With integer field this is cast to nil.

Reply:
was that fixed elsewhere? I didnt see that fix part of this pr?

Reply:
changing the column type to integer fixes it, it saves an empty parameter as `NULL` instead of empty string.